src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/AnnotationFilterPass.php

```diff
@@ -60,6 +61,9 @@ private function createFilterDefinitions(\ReflectionClass $reflectionClass, Read
             $definition->setClass($filterClass);
             $definition->addTag(self::TAG_FILTER_NAME);
             $definition->setAutowired(true);
+            if ((new \ReflectionClass($filterClass))->hasMethod('getPropertyHelperServiceName')) {
+                $definition->setArgument('$propertyHelper', new Reference($filterClass::getPropertyHelperServiceName()));
```

Comment:
Why not something like this?

```php
$definition->setArgument('$propertyHelper', is_a($filterClass, \ApiPlatform\Core\Bridge\Doctrine\MongoDB\Filter::class, true) ? 'the_mongodb_propertyHelper' : 'the_orm_propertyHelper');
```

It's a cleaner design (the classes aren't aware of the DIC, and not tight to Symfony), and it's still extensible (an app or a bundle could use the same trick in its own compiler pass for another provider). 

But here, I think transforming the PropertyHelper class in a trait would be less intrusive.

Reply:
Sure! I don't like the idea of transforming the service into a trait though (my POV explained in the next comment).